File: ruby/basics/word_frequency.rb
#!/usr/bin/env ruby
# Count the words of a fixed text, most frequent first and alphabetically within a tie.

TEXT = 'the quick brown fox jumps over the lazy dog the fox barks'

TEXT.split.tally.sort_by { |word, count| [-count, word] }.each do |word, count|
  puts "#{word} #{count}"
end

File: elixir/basics/word_frequency.exs
# Count the words of a fixed text, most frequent first and alphabetically within a tie.
# Run: elixir word_frequency.exs

text = "the quick brown fox jumps over the lazy dog the fox barks"

text
|> String.split(~r/\s+/, trim: true)
|> Enum.frequencies()
|> Enum.sort_by(fn {word, count} -> {-count, word} end)
|> Enum.each(fn {word, count} -> IO.puts("#{word} #{count}") end)
